### PR TITLE
fix: make iterateTLSConfig only work for TLS config

### PR DESCRIPTION
**Pull request by Ace-Tang**
Signed-off-by: Ace-Tang <aceapril@126.com>

### CODE REVIEW

**Comment by Ace-Tang** [comment]
`iterateConfig` want to solve TLS config at first, but ignore other map kind config like `Runtime` or something, it makes these types of config always be recognized as a unknow flag, to make it compatible, I wanna to make the funtion only work for `TLS` config, and recursive once.

File: daemon/config/config.go [hunk 1]
```diff
@@ -169,7 +169,7 @@ func (cfg *Config) MergeConfigurations(flagSet *pflag.FlagSet) error {
 	}
 
 	fileFlags := make(map[string]interface{}, 0)
-	iterateConfig(origin, fileFlags)
+	iterateTLSConfig(origin, fileFlags)
 
 	// check if invalid or unknown flag exist in config file
 	if err = getUnknownFlags(flagSet, fileFlags); err != nil {
```
Comment (fuweid) [inline daemon/config/config.go]: in your commit, the `UnknownFlags` is only for TLS?

Comment (Ace-Tang) [reply daemon/config/config.go]: `iterateTLSConfig` function is to make TLS value from map become a new key-value
```
{
   "TLS": {
       "a": "b"
    }
}  ->

{
   "a": "b"
}
```
as for other config, they will stay same, so do you got the point? @fuweid

**Comment by allencloud** [comment]
Any update on this? @Ace-Tang @fuweid 
Could we make this move on?

**Pull request merged**